src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -712,20 +490,83 @@ GUID Profile::_GenerateGuidForProfile(const std::wstring& name, const std::optio
 //   (given the `name`)
 // Return Value:
 // - The json's `guid`, or a guid synthesized for it.
-GUID Profile::GetGuidOrGenerateForJson(const Json::Value& json) noexcept
+winrt::guid Profile::GetGuidOrGenerateForJson(const Json::Value& json) noexcept
 {
     if (const auto guid{ JsonUtils::GetValueForKey<std::optional<GUID>>(json, GuidKey) })
     {
-        return guid.value();
+        return { guid.value() };
     }
 
-    const auto name{ JsonUtils::GetValueForKey<std::wstring>(json, NameKey) };
-    const auto source{ JsonUtils::GetValueForKey<std::optional<std::wstring>>(json, SourceKey) };
+    const auto name{ JsonUtils::GetValueForKey<hstring>(json, NameKey) };
+    const auto source{ JsonUtils::GetValueForKey<IReference<hstring>>(json, SourceKey) };
 
     return Profile::_GenerateGuidForProfile(name, source);
 }
 
-void Profile::SetRetroTerminalEffect(bool value) noexcept
+HorizontalAlignment Profile::BackgroundImageHorizontalAlignment() const noexcept
+{
+    if (_BackgroundImageAlignment.has_value())
+    {
+        return std::get<HorizontalAlignment>(_BackgroundImageAlignment.value());
+    }
+    return HorizontalAlignment::Center;
+}
+
+void Profile::BackgroundImageHorizontalAlignment(const HorizontalAlignment& value) noexcept
+{
+    if (!_BackgroundImageAlignment.has_value())
+    {
+        std::get<VerticalAlignment>(_BackgroundImageAlignment.value()) = VerticalAlignment::Center;

```

Comment:
i think this is an illegal way to use an optional.

you're accessing `value()` _even though `has_value` returned false_. that **does not create** the value !!!